### DISCUSSION TITLE
es/datasets/pose/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#13848] [Docs]
# es/datasets/pose/

Learn about Ultralytics YOLO format for pose estimation datasets, supported formats, COCO-Pose, COCO8-Pose, Tiger-Pose, and how to add your own dataset.

https://docs.ultralytics.com/es/datasets/pose/

**Comment by pipeyz21** [comment]
Hi, how are you? 
I wanted to ask if it is possible to write down multiple keypoints for multiple classes.
For example:
class 1: 10 keypoints, class 2: 5 keypoints, class 3: 0
Thank you

**Reply by pderrenger** [reply]
@pipeyz21 hi there! 😊

Thank you for your question. Currently, the Ultralytics YOLO format for pose estimation does not support varying numbers of keypoints for different classes within the same dataset. All classes must have the same number of keypoints defined in the dataset configuration.

However, you can work around this limitation by setting unused keypoints to a default value (e.g., zero) for classes that do not require them. Here’s an example of how you might structure your dataset:

For a dataset with three classes where:
- Class 1 has 10 keypoints
- Class 2 has 5 keypoints
- Class 3 has 0 keypoints

You would define all classes to have the maximum number of keypoints (in this case, 10). For classes with fewer keypoints, you can pad the remaining keypoints with zeros.

Example for class 1 (10 keypoints):
```
<class-index> <x> <y> <width> <height> <px1> <py1> <px2> <py2> ... <px10> <py10>
```

Example for class 2 (5 keypoints, padded with zeros):
```
<class-index> <x> <y> <width> <height> <px1> <py1> <px2> <py2> ... <px5> <py5> 0 0 0 0 0 0 0 0 0 0
```

Example for class 3 (0 keypoints, all zeros):
```
<class-index> <x> <y> <width> <height> 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
```

This way, you can maintain a consistent format across all classes while effectively handling different numbers of keypoints.

For more details on the dataset format and configuration, you can refer to our [Pose Estimation Datasets Overview](https://docs.ultralytics.com/datasets/pose/).

If you have any further questions or need additional assistance, feel free to ask!
